symengine/parser/parser.ih

```diff
@@ -88,6 +102,36 @@ RCP<const Basic> Parser::parse_identifier(const std::string &expr)
     }
 }
 
+#ifdef HAVE_STRTOD_L
+static int strtod_locale_initialized = 0;
+#ifdef _WIN32
+static _locale_t strtod_locale;
+#else
+static locale_t strtod_locale;
+#endif
+#endif
+
+inline double symengine_strtod(const char *startptr, char** endptr) {
+#ifdef HAVE_STRTOD_L
+    if(!strtod_locale_initialized)
+    {
+        strtod_locale_initialized = 1;
+#ifndef _WIN32
+        strtod_locale = newlocale(LC_NUMERIC, "C", NULL);
+#elif !defined(__MINGW32__)
+        strtod_locale = _create_locale(LC_NUMERIC, "C");
+#endif
+    }
+#ifdef _WIN32
+    return _strtod_l(startptr, endptr, strtod_locale);
+#else
+    return strtod_l(startptr, endptr, strtod_locale);
+#endif
+#else
+    return strtod(startptr, endptr);
+#endif
+}
```

Comment:
I think this should probably go into its own header, say `symengine_strtod.h`. So that if we provide our own implementation in the future, it is easy to change.